src/rgb/contract/assignments.rs

```diff
@@ -708,3 +708,813 @@ mod strict_encoding {
+            txid_vec.push(txid);
+        }
+
+        // Take first two amounts to create our allocations
+        let zip_data = txid_vec[..partition]
+            .iter()
+            .zip(output_amounts[..partition].iter());
+
+        // Create our allocations
+        let ours: Vec<(SealDefinition, Amount)> = zip_data
+            .map(|(txid, amount)| {
+                (
+                    Revealed::TxOutpoint(OutpointReveal::from(OutPoint::new(
+                        *txid,
+                        rng.gen_range(0, 10),
+                    ))),
+                    amount.clone(),
+                )
+            })
+            .collect();
+
+        // Take next two amounts for their allocations
+        let zip_data2 = txid_vec[partition..]
+            .iter()
+            .zip(output_amounts[partition..].iter());
+
+        // Create their allocations
+        let theirs: Vec<(OutpointHash, Amount)> = zip_data2
+            .map(|(txid, amount)| {
+                (
+                    Revealed::TxOutpoint(OutpointReveal::from(OutPoint::new(
+                        *txid,
+                        rng.gen_range(0, 10),
+                    )))
+                    .conceal(),
+                    amount.clone(),
+                )
+            })
+            .collect();
+
+        // Balance both the allocations against input amounts
+        let balanced = AssignmentsVariant::zero_balanced(input_revealed.clone(), ours, theirs);
+
+        // Extract balanced confidential output amounts
+        let outputs: Vec<Commitment> = balanced
+            .all_state_pedersen()
+            .iter()
+            .map(|confidential| confidential.commitment)
+            .collect();
+
+        // Create confidential input amounts
+        let inputs: Vec<Commitment> = input_revealed
+            .iter()
+            .map(|revealed| revealed.conceal().commitment)
+            .collect();
+
+        // Check sum verification and return the result
+        amount::Confidential::verify_commit_sum(inputs, outputs)
+    }
+
+    #[test]
+    fn test_zero_balance() {
+        let mut rng = thread_rng();
+
+        // test equal inputs and outputs
+        // 4 values equally distributed among ours and theirs allocations
+        // * One zero value case
+        // * All zero value case
+        // * One u64::MAX value case
+        // * All u64::MAX value case
+        // * One zero value + One u64::MAX value case
+        let amounts = vec![
+            [1u64, 2u64, 3u64, u64::MAX],
+            [10u64, 20u64, 30u64, 40u64],
+            [0u64, 197642u64, u64::MAX, 476543u64],
+            [0u64, 0u64, 0u64, 0u64],
+            [u64::MAX, u64::MAX, u64::MAX, u64::MAX],
+        ];
+
+        for vec in amounts.iter() {
+            assert!(compute_zero_balance(vec, vec, 2));
+        }
+
+        // Test when ours is empty
+        assert!(compute_zero_balance(&amounts[1], &amounts[1], 0));
```

Comment:
Are you sure about this one? This will simply go "index out of bound error". Also it won't test "ours only" which will only happen with `partition=0`.

Reply:
sorry, what I meant was to use `amounts[10]` instead of `amounts[1]`

Reply:
Got it.